src/cow-react/modules/trade/hooks/useTradeNavigate.ts

```diff
@@ -0,0 +1,26 @@
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { useHistory } from 'react-router-dom'
+import { useCallback } from 'react'
+import { useTradeTypeInfo } from '@cow/modules/trade/hooks/useTradeTypeInfo'
+import { TradeCurrenciesIds } from '@cow/modules/trade/types/TradeState'
+import { parameterizeTradeRoute } from '@cow/modules/trade/utils/parameterizeTradeRoute'
+
+interface UseTradeNavigateCallback {
+  (chainId: SupportedChainId | null | undefined, { inputCurrencyId, outputCurrencyId }: TradeCurrenciesIds): void
+}
+
+export function useTradeNavigate(): UseTradeNavigateCallback {
+  const history = useHistory()
+  const tradeTypeInfo = useTradeTypeInfo()
+
+  return useCallback(
+    (chainId: SupportedChainId | null | undefined, { inputCurrencyId, outputCurrencyId }: TradeCurrenciesIds) => {
```

Comment:
Same nitpick here: possible to re-use the callback type?

Reply:
This is function body